src/core/components/files.js

```diff
@@ -282,5 +282,22 @@ module.exports = function files (self) {
     },
 
     lsPullStreamImmutable: _lsPullStreamImmutable
-  }
+  },
+
+  unlink: promisify((ipfsPath, callback) => {
+    const cid = new CID(ipfsPath)
+    self._ipldResolver.get(cid, (err, node) => {
+      if (err) {
+        return callback(err)
+      }
+      pull(
+        pull.values(node.value.links),
+        pull.drain(
+          link => self._ipldResolver.bs.delete(new CID(link.multihash)),
+          () => self._ipldResolver.bs.delete(cid, callback)
+        )
+      )
```

Comment:
This will delete 1 level of links, but if the links have links then they should also be deleted as per `pin rm --recursive`.